test/EngineTest.cpp

```diff
@@ -23,114 +32,259 @@ auto I = [](const auto& id) {
-  b.push_back({I(3), I(1)});
-  b.push_back({I(4), I(2)});
-  IdTable res(3, allocator());
-  int lwidth = a.numColumns();
-  int rwidth = b.numColumns();
-  int reswidth = a.numColumns() + b.numColumns() - 1;
-  Join J{Join::InvalidOnlyForTestingJoinTag{}};
-  CALL_FIXED_SIZE((std::array{lwidth, rwidth, reswidth}), &Join::join, J, a, 0,
-                  b, 0, &res);
-
-  ASSERT_EQ(I(1), res(0, 0));
-  ASSERT_EQ(I(1), res(0, 1));
-  ASSERT_EQ(I(3), res(0, 2));
-  ASSERT_EQ(I(1), res(1, 0));
-  ASSERT_EQ(I(1), res(1, 1));
-  ASSERT_EQ(I(8), res(1, 2));
-  ASSERT_EQ(I(1), res(2, 0));
-  ASSERT_EQ(I(3), res(2, 1));
-  ASSERT_EQ(I(3), res(2, 2));
-  ASSERT_EQ(I(1), res(3, 0));
-  ASSERT_EQ(I(3), res(3, 1));
-  ASSERT_EQ(I(8), res(3, 2));
-  ASSERT_EQ(5u, res.size());
-  ASSERT_EQ(I(4), res(4, 0));
-  ASSERT_EQ(I(1), res(4, 1));
-  ASSERT_EQ(I(2), res(4, 2));
-
-  res.clear();
-  for (size_t i = 1; i <= 10000; ++i) {
-    b.push_back({I(4 + i), I(2 + i)});
+/*
+ * Return an 'IdTable' with the given 'tableContent'. all rows must have the
+ * same length.
+*/
+IdTable makeIdTableFromVector(std::vector<std::vector<size_t>> tableContent) {
+  IdTable result(tableContent[0].size(), allocator());
+
+  // Copying the content into the table.
+  for (const auto& row: tableContent) {
+    const size_t backIndex = result.size();
+    result.emplace_back();
+
+    for (size_t c = 0; c < tableContent[0].size(); c++) {
+      result(backIndex, c) = I(row[c]);
+    }
   }
-  a.push_back({I(400000), I(200000)});
-  b.push_back({I(400000), I(200000)});
 
-  CALL_FIXED_SIZE((std::array{lwidth, rwidth, reswidth}), &Join::join, J, a, 0,
-                  b, 0, &res);
-  ASSERT_EQ(6u, res.size());
+  return result;
+}
 
-  a.clear();
-  b.clear();
-  res.clear();
+/*
+ * A structure containing all information needed for a normal join test. A
+ * normal join test is defined as two IdTables being joined, the resulting
+ * IdTable tested, if it is sorted after the join column, or not, and this
+ * IdTable than compared with the sampleSolution.
+ */
+struct normalJoinTest {
+  IdTable leftIdTable;
+  size_t leftJC; // Join column for the left IdTable.
+  IdTable rightIdTable;
+  size_t rightJC; // Join column for the right IdTable.
+  bool testForSorted; // Should the result of the join be tested, if it is
+                      // orderd after leftJC? In other words, ordered after the join column?
+  IdTable sampleSolution;
+
+  // The initialization list is needed, because IdTables can't be constructed
+  // without arguments.
+  normalJoinTest(const IdTable& LeftIdTable,
+      const size_t LeftJC,
+      const IdTable& RightIdTable,
+      const size_t RightJC,
+      const IdTable& SampleSolution,
+      const bool TestForSorted = false): leftIdTable(LeftIdTable.clone()),
+      leftJC(LeftJC),
+      rightIdTable(RightIdTable.clone()),
+      rightJC(RightJC),
+      testForSorted(TestForSorted),
+      sampleSolution(SampleSolution.clone())
+  {}
```

Comment:
I created this contructor, so that generating a `normalJoinTest`, without knowing about the specifics of the `IdTable` copy constructor, was easy.

But if you don't think, that this help is needed, then I will bow to your wisdom. Done in [ec21f8f](https://github.com/ad-freiburg/qlever/pull/821/commits/ec21f8fe96e7e8a1e80e189df2c0d3d9042cb301).